lib/fetch-template.js

```diff
@@ -26,44 +26,78 @@ class TemplateError extends Error {
  *
  * @param {string} path
  */
-const readFile = (path) => 
+const readFile = (path) =>
     new Promise((resolve, reject) => {
         fs.readFile(path, 'utf-8', (err, data) => {
             if (err) {
                 reject(new TemplateError(err));
                 return;
-            } 
+            }
             resolve(data);
         });
     });
 
 /**
+ * Returns the template path validating a exactly file or a directory
+ *
+ * @param  {Object} request - Request Object
+ * @param  {String} templatesPath - TemplatesPath config
+ *
+ * @return {Promise} Template Path on success or TemplateError on fail
+ */
+const getTemplatePath = (request, templatesPath) => new Promise((resolve, reject) => {
+    fs.lstat(
```

Comment:
If you're using node 8.x, you can use [`util.promisify`](https://nodejs.org/api/util.html#util_util_promisify_original) to make this code simpler